[CKLS-21158] Replaced the symfony/symfony package with the symfony/flex

Alex-Caepio opened the pull request.

Alex-Caepio committed:
[CKLS-21158] Replaced the symfony/symfony package with the symfony/flex

Alex-Caepio committed:
[CKLS-21158] Replaced the symfony/symfony package with the symfony/flex

Alex-Caepio committed:
[CKLS-21158] Replaced the symfony/symfony package with the symfony/flex

Alex-Caepio committed:
[CKLS-21158] Replaced the symfony/symfony package with the symfony/flex

irena-zaz commented on composer.json:
```diff
@@ -20,11 +20,32 @@
   },
   "require": {
     "php": ">=5.5",
```
```suggestion
    "php": ">=7.2.5",
```
we should use compatible version with Symfony 5.4

Alex-Caepio replied:
Upgraded the PHP version to 7.2.5

irena-zaz commented:
We do not run the unit tests, in the pipelines, will you confirm that they all pass?

Alex-Caepio committed:
[CKLS-21158] Update PHP requirement to >=7.2.5 and bump dependencies

Updated `composer.json` to require PHP >=7.2.5, aligning with modern PHP versions. Also upgraded `symfony/flex` to v1.22.0 and `myclabs/deep-copy` to v1.13.0, ensuring compatibility and access to the latest features and fixes. Adjusted `composer.lock` to reflect these changes.
